File: Bai7.py
import math

def tim_ucln(a, b):
    try:
        a, b = int(a), int(b)
    except ValueError:
        return "Lỗi: Đầu vào phải là số nguyên."
        
    if a == 0 and b == 0:
        return "Lỗi: a và b không thể cùng bằng 0."
        
    return math.gcd(abs(a), abs(b))

print("--- KẾT QUẢ CHẠY TEST CASE BÀI 7 ---")
print(f"TC01 (12, 8): {tim_ucln(12, 8)}")
print(f"TC02 (-15, 5): {tim_ucln(-15, 5)}")
print(f"TC03 (0, 7): {tim_ucln(0, 7)}")
print(f"TC04 (0, 0): {tim_ucln(0, 0)}")
print(f"TC05 ('x', 2): {tim_ucln('x', 2)}")

File: Bai7.java
public class Bai7 {
    public static int timUCLN(int a, int b) {
        if (a == 0 && b == 0) throw new IllegalArgumentException("Không xác định");
        a = Math.abs(a); b = Math.abs(b);
        while (b != 0) {
            int temp = a % b;
            a = b;
            b = temp;
        }
        return a;
    }
}